src/services/doppler-api-client.ts

```diff
@@ -0,0 +1,78 @@
+import { ResultWithoutExpectedErrors } from '../doppler-types';
+import { AxiosInstance, AxiosStatic } from 'axios';
+import { AppSession, DopplerAPIConnectionData } from './app-session';
+import { RefObject } from 'react';
+import { SubscriberList, SubscriberListState } from './shopify-client';
+
+export interface DopplerAPIClient {
+  getListData(idList: number, apikey: string): Promise<ResultWithoutExpectedErrors<SubscriberList>>;
+}
+
+export class HttpDopplerAPIClient implements DopplerAPIClient {
+  private readonly axios: AxiosInstance;
+  private readonly baseUrl: string;
+  private readonly connectionDataRef: RefObject<AppSession>;
+
+  constructor({
+    axiosStatic,
+    baseUrl,
+    connectionDataRef,
+  }: {
+    axiosStatic: AxiosStatic;
+    baseUrl: string;
+    connectionDataRef: RefObject<AppSession>;
+  }) {
+    this.baseUrl = baseUrl;
+    this.axios = axiosStatic.create({
+      baseURL: this.baseUrl,
+    });
+    this.connectionDataRef = connectionDataRef;
+  }
+
+  private getDopplerAPIConnectionData(): DopplerAPIConnectionData {
+    const connectionData = this.connectionDataRef.current;
+    if (
+      !connectionData ||
+      connectionData.status !== 'authenticated' ||
+      !connectionData.jwtToken ||
+      !connectionData.userData
+    ) {
+      throw new Error('Doppler API connection data is not available');
+    }
+    return { jwtToken: connectionData.jwtToken, userAccount: connectionData.userData.user.email };
```

Comment:
This method structure is different than other related implementations:

https://github.com/FromDoppler/doppler-webapp/blob/30cbe6fdcb9a4216360ba001025aa9dabb6b0982/src/services/shopify-client.ts#L49-L55

https://github.com/FromDoppler/doppler-webapp/blob/30cbe6fdcb9a4216360ba001025aa9dabb6b0982/src/services/datahub-client.ts#L68-L78

In that cases, the resulting object is the same connection data, but casted with the expected structure.

I understand the convenience of this method, so I am fine on using in this way.

Only two things:

* `DopplerAPIConnectionData` type should be scoped because it is not used in other places.

* Maybe you could use deconstruction here:

    ```suggestion
        return { connectionData: { jwtToken }, connectionData: { userData: { user: { userAccount: email } } } };
    ```

Reply:
I don't see this work as expected 😕

Reply:
Do not worry, I am not so much _canchero_ with deconstructing.